lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -19,120 +20,186 @@
+    $dest_parent = Split-Path -LiteralPath $dest
+    if (-not (Test-Path -LiteralPath $dest_parent -PathType Container)) {
+        Fail-Json -obj $result -message "The path '$dest_parent' does not exist for destination '$dest', or is not visible to the current user.  Ensure download destination folder exists (perhaps using win_file state=directory) before win_get_url runs."
+    }
+
+    $webClient = New-Object System.Net.WebClient
+
+    foreach ($header in $headers.GetEnumerator()) {
+        $webClient.Headers.Add($header.Name, $header.Value)
+    }
+
+    if ($timeout) {
+        $webClient.Timeout = $timeout * 1000
+    }
+
+    if (-not $use_proxy) {
+        $webClient.Proxy = $null
+    } elseif ($proxy) {
+        $webClient.Proxy = $proxy
+    }
+
+    if ($credentials) {
+        $webClient.Credentials = $credentials
+    }
+
+    Try {
+        if (-not $whatif) {
+            $webClient.DownloadFile($url, $dest)
+        }
+        $result.changed = $true
+    } Catch [System.Net.WebException] {
+        $result.status_code = $_.Exception.Response.StatusCode
+        Fail-Json -obj $result -message "Error downloading '$url' to '$dest'. $($_.Exception.Message)"
+    } Catch {
+        Fail-Json -obj $result -message "Unknown error downloading '$url' to '$dest'. $($_.Exception.Message)"
+    }
+    # FIXME: Reimplement DownloadFile() so we get the real information
+    $result.status_code = 200
+    $result.msg = 'OK'
+    $result.dest = $dest
+}
+
+
 $params = Parse-Args $args -supports_check_mode $true
 $check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
 
 $url = Get-AnsibleParam -obj $params -name "url" -type "str" -failifempty $true
 $dest = Get-AnsibleParam -obj $params -name "dest" -type "path" -failifempty $true
+$timeout = Get-AnsibleParam -obj $params -name "timeout" -type "int" -default 10
+$headers = Get-AnsibleParam -obj $params -name "headers" -type "dict"
 $skip_certificate_validation = Get-AnsibleParam -obj $params -name "skip_certificate_validation" -type "bool"
 $validate_certs = Get-AnsibleParam -obj $params -name "validate_certs" -type "bool" -default $true
-$username = Get-AnsibleParam -obj $params -name "username" -type "str"
-$password = Get-AnsibleParam -obj $params -name "password" -type "str"
+$url_username = Get-AnsibleParam -obj $params -name "url_username" -type "str" -aliases "username"
+$url_password = Get-AnsibleParam -obj $params -name "url_password" -type "str" -aliases "password"
+$use_proxy = Get-AnsibleParam -obj $params -name "use_proxy" -type "bool" -default $true
 $proxy_url = Get-AnsibleParam -obj $params -name "proxy_url" -type "str"
 $proxy_username = Get-AnsibleParam -obj $params -name "proxy_username" -type "str"
 $proxy_password = Get-AnsibleParam -obj $params -name "proxy_password" -type "str"
 $force = Get-AnsibleParam -obj $params -name "force" -type "bool" -default $true
 
 $result = @{
     changed = $false
-    win_get_url = @{
```

Comment:
I still don't like the idea of us just taking this away especially for a core module. I think we should keep it for 1-2 major releases and just document it will be deprecated at 2.x. That way people have a bridging point to swap from one variable to another instead of them needing to change their scripts.